src/Network/Network/NetworkWatcher/ConnectionMonitor/ConnectionMonitorBaseCmdlet.cs

```diff
@@ -88,5 +109,637 @@ public MNM.NetworkWatcher GetNetworkWatcherByLocation(string location)
+                return null;
+
+            foreach (ConnectionMonitorTestConfiguration TestConfiguration in TestConfigurations)
+            {
+                if (string.Compare(TestConfigName, TestConfiguration.Name) == 0)
+                {
+                    return TestConfiguration;
+                }
+            }
+
+            return null;
+        }
+
+        public MNM.ConnectionMonitorEndpoint GetEndpoinByName(IList<ConnectionMonitorEndpoint> Endpoints, string EndpointName)
+        {
+            if (Endpoints == null || string.IsNullOrEmpty(EndpointName))
+                return null;
+
+            foreach (ConnectionMonitorEndpoint Endpoint in Endpoints)
+            {
+                if (string.Compare(EndpointName, Endpoint.Name) == 0)
+                {
+                    return Endpoint;
+                }
+            }
+
+            return null;
+        }
+
+    public bool ValidateConnectionMonitorV1V2Parameters(string SourceResourceId, string DestinationResourceId, PSConnectionMonitorResult InputObject, string DestinationAddress, int? MonitoringIntervalInSeconds, List<PSNetworkWatcherConnectionMonitorTestGroupObject> TestGroups, List<PSNetworkWatcherConnectionMonitorOutputObject> Outputs)
+        {
+            if ((!string.IsNullOrEmpty(SourceResourceId)     || 
+                 !string.IsNullOrEmpty(DestinationResourceId)||
+                 !string.IsNullOrEmpty(DestinationAddress)   ||
+                 MonitoringIntervalInSeconds != null) && (TestGroups != null || InputObject !=null))
+            {
+                throw new ArgumentException("Connection moniotr V1 can not be defined with either TestGroup. Either connection monitor V1 or V2 can be specified");
+            }
+
+            if (string.IsNullOrEmpty(SourceResourceId) && InputObject == null && TestGroups == null)
+            {
+                throw new ArgumentException("Either SourceResourceId or InputObject or TestGroups is to be ddefined");
+            }
+
+            // Validate Test Group
+            if (TestGroups != null && TestGroups.Any())
+            {
+                foreach (PSNetworkWatcherConnectionMonitorTestGroupObject TestGroup in TestGroups)
+                {
+                    // validate mandatory parameters
+                    if (string.IsNullOrEmpty(TestGroup.Name) || TestGroup.TestConfigurations == null || TestGroup.Sources == null || TestGroup.Destinations == null)
+                    {
+                        throw new ArgumentException("Test group is missing one or more mandatory parameter");
+                    }
+
+                    // validate Source and Destination Endpoints
+                    List<PSNetworkWatcherConnectionMonitorEndpointObject> Endpoints = TestGroup.Sources;
+                    Endpoints.Concat(TestGroup.Destinations);
+
+                    //validate end point
+                    if (Endpoints.Any())
```

Comment:
I think this code might be easier to read if you split up the validation into separate validation methods (Endpoints, V1, V2, etc.)

Reply:
But really I more toward optimization. if we have two parts; it will be reptitive code and I want to avoid it. It is the same check that is done on all endpoints whether they are are source or destination endpoints

Reply:
I  did that for optimization. Making two checks will repeat the same code twice and thus I did as as one block

Reply:
Done